jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.c

```diff
@@ -1131,11 +1104,13 @@ ecma_builtin_string_prototype_object_replace_loop (ecma_builtin_replace_search_c
       {
         /* No more matches */
         ecma_string_t *appended_string_p;
+
+        ecma_ref_ecma_string (result_string_p);
```

Comment:
I don't see why is this ref needed here. Can you explain it?

Reply:
There is an `ecma_deref_ecma_string (result_string_p);` below. Actually this part of the code should be rewritten as a whole, but I felt it could be done in a follow-up patch.

Reply:
Yes, it should be done in a follow-up patch. Thanks for the explanation.